File: Fortran/Recap/Reduction/working_reduce.f90
PROGRAM reduce

  USE mpi
  IMPLICIT NONE

  INTEGER, PARAMETER :: tag = 100

  INTEGER :: rank, recv
  INTEGER :: nproc, ierr

  CALL MPI_Init(ierr)

  CALL MPI_Comm_size(MPI_COMM_WORLD, nproc, ierr)
  CALL MPI_Comm_rank(MPI_COMM_WORLD, rank, ierr)

  CALL MPI_Reduce(rank, recv, 1, MPI_INTEGER, MPI_MIN, 0, MPI_COMM_WORLD, ierr)
  IF (rank == 0) PRINT *, ' Minimum value of rank is ', recv

  CALL MPI_Reduce(rank, recv, 1, MPI_INTEGER, MPI_MAX, 0, MPI_COMM_WORLD, ierr)
  IF (rank == 0) PRINT *, ' Maximum value of rank is ', recv

  CALL MPI_Reduce(rank, recv, 1, MPI_INTEGER, MPI_SUM, 0, MPI_COMM_WORLD, ierr)
  IF (rank == 0) PRINT *, ' Sum of all ranks is ', recv

  CALL MPI_Finalize(ierr)

END PROGRAM reduce

File: C/Recap/Reduction/working_reduce.c
#include <stdio.h>
#include <mpi.h>

#define TAG 100

int main(int argc, char ** argv)
{

  int rank, nproc, recv;

  MPI_Init(&argc, &argv);
  MPI_Comm_size(MPI_COMM_WORLD, &nproc);
  MPI_Comm_rank(MPI_COMM_WORLD, &rank);

  MPI_Reduce(&rank, &recv, 1, MPI_INT, MPI_MIN, 0, MPI_COMM_WORLD);
  if (rank == 0) printf("Minimum value of rank is %3d\n", recv);

  MPI_Reduce(&rank, &recv, 1, MPI_INT, MPI_MAX, 0, MPI_COMM_WORLD);
  if (rank == 0) printf("Maximum value of rank is %3d\n", recv);

  MPI_Reduce(&rank, &recv, 1, MPI_INT, MPI_SUM, 0, MPI_COMM_WORLD);
  if (rank == 0) printf("Sum of all ranks is %3d\n", recv);

  MPI_Finalize();

}
